src/codegen/doc/format.rs

```diff
@@ -0,0 +1,69 @@
+use std::ascii::AsciiExt;
+use regex::{Captures, Regex};
+
+const LANGUAGE_SEP_BEGIN : &'static str = "<!-- language=\"";
+const LANGUAGE_SEP_END : &'static str = "\" -->";
+const LANGUAGE_BLOCK_BEGIN : &'static str = "|[";
+const LANGUAGE_BLOCK_END : &'static str = "\n]|";
+
+pub fn reformat_doc(input: &str, namespace_name: &str) -> String {
```

Comment:
No but I got the struct name wrong: `analysis::namespaces::Namespace` is the one which has `crate_name`